os/android/libhwcservice/test/hwcservice_test.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include <binder/IServiceManager.h>
+#include <binder/ProcessState.h>
+#include <binder/TextOutput.h>
+#include <utils/String8.h>
+#include <cinttypes>
+#include "hwcserviceapi.h"
+#include "iservice.h"
+
+#define LOG_TAG "hwcservice"
+#include <cutils/log.h>
+
+using namespace android;
+using namespace hwcomposer;
+
+static void usage() {
+  aout << "Usage: hwcservice_test \n"
+               "\t-g: Get current display mode\n"
+               "\t-s: Set display mode\n"
+               "\t-p: Print all available display modes\n"
+               "\t-u: Set Hue\n";
+   exit(-1);
+}
+
+int main(int argc, char** argv) {
+  uint32_t display = 0;
+  uint32_t display_mode_index = 0;
+  bool print_mode = false;
+  bool get_mode = false;
+  bool set_mode = false;
+  bool set_hue = false;
+  int ch;
+  while ((ch = getopt(argc, argv, "gsphu")) != -1) {
+    switch (ch) {
+      case 'g':
+        get_mode = true;
+        break;
+      case 's':
+        set_mode = true;
+        break;
+      case 'p':
+        print_mode = true;
+        break;
+      case 'u':
+        set_hue = true;
+        break;
+      case 'h':
+      default:
+        usage();
+    }
+  }
+  argc -= optind;
+  argv += optind;
+
+  // Initialize ProcessState with /dev/vndbinder as HwcService is in vndbinder context
+  android::ProcessState::initWithDriver("/dev/vndbinder");
```

Comment:
Move this under USE_PROCESS_STATE checks

Reply:
Where is this macro defined? Is it defined by default under O?

Reply:
Ya, https://github.com/intel/IA-Hardware-Composer/blob/master/Android.common.mk#L126